### PR TITLE
Spring 11 magdy

### PR DESCRIPTION
**Pull request by ahmedhem**

### CODE REVIEW

File: src/Canvas/canvasFunctions.js [hunk 1]
```diff
@@ -79,88 +65,107 @@ function drawWeight(ctx, points, weight) {
   ctx.fillStyle = "#111";
   ctx.closePath();
 }
-export function drawEdge(ctx, node1, node2, size, color = null) {
+
+/*
```
Comment (ahmedEid1) [inline src/Canvas/canvasFunctions.js:69]: great explanation, thanks 👍

File: src/Canvas/canvasFunctions.js [hunk 2]
```diff
@@ -79,88 +65,107 @@ function drawWeight(ctx, points, weight) {
...
+  /************/
+  // second node
+  p = tranlsate_point(x, y, calcSlope(x, y, x1, y1), size, 1);
+  p2 = tranlsate_point(x, y, calcSlope(x, y, x1, y1), size, -1);
+  [x_move, y_move] = getDist(x1, y1, p[0], p[1]) < getDist(x, y, x1, y1) ? p : p2;
+
+  return [x_move, y_move, x1_move, y1_move];
+};
+
+export async function drawEdge(ctx, node1, node2, size, color = null) {
```
Comment (ahmedEid1) [inline src/Canvas/canvasFunctions.js]: why does drawEdge need to be an async function?!

File: src/Canvas/canvasFunctions.js [hunk 3]
```diff
@@ -79,88 +65,107 @@ function drawWeight(ctx, points, weight) {
...
+}
+//https://math.stackexchange.com/a/409737
+export function tranlsate_point(x, y, slope, d, dir) {
+  let x_move = x + dir * d / Math.sqrt(1 + slope * slope);
+  let y_move = y + slope * (x_move - x);
+  return [x_move, y_move];
+}
+
+/*
+drawiung a curve must have three points , start point , end point , controlling point
```
Comment (ahmedEid1) [inline src/Canvas/canvasFunctions.js:131]: 👍 Nice!!

File: src/Visualizer/Visualizer.js [hunk 4]
```diff
@@ -32,21 +40,22 @@ class Visualizer {
...
 
   change_node_size(new_size) {
     let Node = new_size.node;
-    Node.size = new_size.size
-
+    Node.size = new_size.size;
+    new_size.animate();
   }
-
+  // new edge color will by only chaned in the Graph  and won't be animated (as it doesn't make sense).
```
Comment (ahmedEid1) [inline src/Visualizer/Visualizer.js:130]: the change of the color should be animated on the canvas...we already have an animation for it

File: src/Visualizer/Visualizer.js [hunk 5]
```diff
@@ -32,21 +40,22 @@ class Visualizer {
   change_node_position(new_position) {
     let Node = new_position.node;
     Node.position = new_position.position;
+    new_position.animate();
   }
-
+  // new node color will by only chaned in the Graph  and won't be animated
```
Comment (ahmedEid1) [inline src/Visualizer/Visualizer.js:94]: I agree for now...but maybe in the future, we can create a good animation for it?

Comment (ahmedhem) [reply src/Visualizer/Visualizer.js:94]: Right now I don't have any idea why and how to animate the color of the node, so we can leave it for now until we have a reason to animate the node color.

File: src/Canvas/canvasFunctions.js (hunk 2 above)
Comment (ahmedhem) [reply src/Canvas/canvasFunctions.js]: it was kind of testing for the async method and I forgot to delete it after that.
sorry for the inconvenience.

**Pull request merged**